tests/ui/consts/const-eval/issue-91827-extern-types-field-offset.rs

```diff
@@ -0,0 +1,32 @@
+// Test that we can handle unsized types with an extern type tail part.
+// Regression test for issue #91827.
+
+#![feature(extern_types)]
+
+use std::ptr::addr_of;
+
+extern "C" {
+    type Opaque;
+}
+
+struct Newtype(Opaque);
+
+struct S {
+    i: i32,
+    a: Opaque,
+}
+
+const NEWTYPE: () = unsafe {
+    // Projecting to the newtype works, because it is always at offset 0.
+    let x: &Newtype = unsafe { &*(1usize as *const Newtype) };
+    let field = &x.0;
```

Comment:
```suggestion
    let x: &Newtype = unsafe { &*(1usize as *const Newtype);

    // Projecting to the newtype works, because it is always at offset 0.
    let field = &x.0;
```